compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -2531,12 +2531,13 @@ static bool virtualGuardHelper(TR::Node *node, TR::CodeGenerator *cg)
    TR::LabelSymbol *label = node->getBranchDestination()->getNode()->getLabel();
 
    cg->setVMThreadRequired(true);
+
    TR::Instruction *vgnopInstr = generateVirtualGuardNOPInstruction(node, site, deps, label, cg);
    TR::Instruction *patchPoint = cg->getVirtualGuardForPatching(vgnopInstr);
 
-   // HCR guards are patched when the threads are stopped.
-   // No issues with multithreaded patching, therefore alignment is not required
-   if (TR::Compiler->target.isSMP() && !node->isHCRGuard() && !node->isOSRGuard())
+   // Guards patched when the threads are stopped have no issues with multithreaded patching. 
+   // therefore alignment is not required
+   if (TR::Compiler->target.isSMP() && !cg->isStopTheWorldGuard(node))
```

Comment:
Is a similar change needed in any other codegens?

Reply:
Thanks @fjeremic can you review the z change?

Reply:
The other codegen doesn't need patching alignment but after discussion with @fjeremic there is a binary encoding opt that could be applied to breakpoint guard as it is a stop the world guard. I am changing that code to use this query.